Sources/SwiftIdentifier.swift

```diff
@@ -78,5 +74,19 @@ func swiftIdentifier(from string: String,
       return ""
     }
   })
-  return prefix + mappedParts.joined(separator: replacement)
+
+  let result = mappedParts.joined(separator: replacement)
+  return prefixWithUnderscoreIfNeeded(string: result, forbiddenChars: exceptions)
+}
+
+func prefixWithUnderscoreIfNeeded(string: String,
+                                  forbiddenChars exceptions: String = "") -> String {
+
+  let (head, tail) = identifierCharacterSets(exceptions: exceptions)
+
+  let chars = string.unicodeScalars
+  let firstChar = chars[chars.startIndex]
+  let prefix = !head.longCharacterIsMember(firstChar.value) && tail.longCharacterIsMember(firstChar.value) ? "_" : ""
```

Comment:
That bit of code used to be in the `swiftIdentifier(from:...)` method, and it had to check if it was still a valid character, otherwise that character was going to be replaced by a `_`.

I think (not 100% sure) that the check for tail can be removed here.